src/problems/chinese_ring_puzzle.clj

```diff
@@ -0,0 +1,42 @@
+; we used the two rules of chinese ring puzzle:
+; 1) the first ring, which is hooked in the very front of the loop, can be taken off or put back on the loop any time;
+; 2) the only other ring that can be taken off from or put back on the loop is the ring next to the first ring.
+(ns problems.chinese-ring-puzzle)
+
+(defrecord State [leadRing Rings])
+
+(defn children [state]
```

Comment:
This is really long (for a Clojure method) and probably could be broken up some.